jwst/resample/resample_step.py

```diff
@@ -170,12 +171,20 @@ def _check_list_pars(vals, name, min_vals=None):
             raise ValueError(f"Both '{name}' values must be either None or not None.")
 
     @staticmethod
-    def _load_custom_wcs(asdf_wcs_file, output_shape):
-        if not asdf_wcs_file:
+    def _load_custom_wcs(input_wcs_file, output_shape):
+        if not input_wcs_file:
             return None
 
-        with asdf.open(asdf_wcs_file) as af:
-            wcs = deepcopy(af.tree["wcs"])
+        try:
+            # File is ASDF file with a top-level wcs in the tree
+            with asdf.open(input_wcs_file) as af:
+                wcs = deepcopy(af.tree["wcs"])
```

Comment:
Would it be too much burden to require the `wcs` attribute in an asdf file to be under `meta` as well, to make it uniform? An argument for this is that this pipeline is supposed to work equivalently with asdf files, so it's conceivable some time in the future the reference file will be an `i2d.asdf` file with the wcs under `model.meta.wcs`. 
I think the uniformity of the file structure may help users, though if it's considered too much of a burden, I'd be OK with the current proposal.

Reply:
I think that would be an API change, as up to this point it's been required to be at the top level.  I.e. this PR doesn't change that.  I do know there's existing analysis code in the wild (ERO CEERS code on github) that specifically uses this functionality.

Agree it would be good to have consistency.  Maybe better to move `wcs` out of `meta` instead, something we've talked about off and on before.